frame/support/procedural/src/construct_runtime/expand/inherent.rs

```diff
@@ -110,9 +138,9 @@ pub fn expand_outer_inherent(
 								if #pallet_names::is_inherent(call) {
 									is_inherent = true;
 									if let Err(e) = #pallet_names::check_inherent(call, self) {
-										result.put_error(
+										handle_put_error_result(result.put_error(
```

Comment:
I would suggest that we move the `if fata_error { return }` part also to a helper function as this is the part that is still important and ensures we bail after seeing one error.

Reply:
I'm not exactly sure how to factorise it out. something like this?
```diff
@@ -138,10 +138,12 @@ pub fn expand_outer_inherent(
                                                                if #pallet_names::is_inherent(call) {
                                                                        is_inherent = true;
                                                                        if let Err(e) = #pallet_names::check_inherent(call, self) {
-                                                                               handle_put_error_result(result.put_error(
-                                                                                       #pallet_names::INHERENT_IDENTIFIER, &e
+                                                                               let is_fatal = put_error_and_is_error_fatal(
+                                                                                       result,
+                                                                                       #pallet_names::INHERENT_IDENTIFIER,
+                                                                                       &e
                                                                                ));
-                                                                               if e.is_fatal_error() {
+                                                                               if is_fatal {
                                                                                        return result;
                                                                                }
                                                                        }
```